### PR TITLE
Dump table (effil.table -> table)

### LINKED ISSUE #132: Method to turn an effil.table into a normal table
Hi and thanks for `effil`!

Sometimes it's be nice to be able to turn an `effil.table` back into a normal `table`, for example when using external libraries that check for the `type()` of an input.

One can do this with a simple loop, bu it's a bit tedious (especially for nested tables). Can you provide a function to do this? 

(if it's not there already and I missed it!)

Thanks!

### PR DESCRIPTION
**Pull request by mihacooper**
closes #132

### CODE REVIEW

File: src/cpp/stored-object.cpp [hunk 1]
```diff
@@ -94,6 +94,15 @@ class GCObjectHolder : public BaseHolder {
     sol::optional<T> strongRef_;
 };
 
+class SharedTableHolder : public GCObjectHolder<SharedTable> {
+public:
+    using GCObjectHolder<SharedTable>::GCObjectHolder;
+
+    sol::object convertToNative(sol::this_state state, DumpCache& cache) const final {
```
Comment (IUdalov) [inline src/cpp/stored-object.cpp]: It's a bit strange naming. I suggest `confvertToLua`. `native` can refer either to C++ native code or native language - Lua.

File: src/cpp/function.h [hunk 2]
```diff
@@ -18,9 +18,13 @@ class FunctionData : public GCData {
 
 class Function : public GCObject<FunctionData> {
 public:
-    sol::object loadFunction(lua_State* state);
+    sol::object loadFunction(lua_State* state) const;
+    sol::object convertToNative(lua_State* state, BaseHolder::DumpCache& cache) const;
 
 private:
+    using ConvertClbk = std::function<sol::object(const StoredObject&)>;
```
Comment (IUdalov) [inline src/cpp/function.h]: This function can be named as `Converter`.

**Review by IUdalov — APPROVED** [review]
LGTM

File: src/cpp/stored-object.cpp (hunk 1 above)
Comment (mihacooper) [reply src/cpp/stored-object.cpp]: ok

File: src/cpp/function.h (hunk 2 above)
Comment (mihacooper) [reply src/cpp/function.h]: ok

**Pull request merged**